### ISSUE TITLE
Enum serialization error on JS, while working on JVM

### ISSUE THREAD

**Issue by Coneys** [Kotlin/kotlinx.serialization#858]
**Describe the bug**
Serialization on JS throws error if we have enum in constructor and we read it in class body
**To Reproduce**

Run test below on JVM and JS. JVM works fine, while JS throws exception about enum.name:
TypeError: Cannot read property 'name' of undefined

```
class JsSerializationErrorPoC {
    @Test
    fun testFunc() {
        val json = Json(JsonConfiguration.Stable.copy(useArrayPolymorphism = false))

        val string = json.stringify(ClassWithEnum.serializer(), ClassWithEnum(SomeEnum.A))

        println(string)

    }
}

enum class SomeEnum {
    A;
}

@Serializable
class ClassWithEnum(val enum: SomeEnum){
    val enumName = enum.name
}
```

**Expected behavior**
If there is no way for this code to work on some platform, plugin should warn us about this. If possible, it should work. Its not hard to workaround that problem (simple get() will do) but I wasted some time searching for bug

**Environment**
 - Kotlin version: [e.g. 1.3.70]
 - Library version: [e.g. 0.20.0]
 - Kotlin platforms: [e.g. JVM, JS]
 - Gradle version: [e.g. 6.1.1]

**Comment by shanshin** [comment]
Not reproduced on latest versions: Kotlin `1.5.30`, library `1.2.2`.

**Comment by shanshin** [comment]
Fixed in JS/IR (checked for Kotlin `1.5.30` library `1.2.2`)
